File: 2026/05-May/25-Monday/solution.js
var canReach = function(s, minJump, maxJump) {

    let n = s.length;
    let dp = Array(n).fill(false);

    dp[0] = true;
    let reachable = 0;

    for(let i=1;i<n;i++){

        if(i >= minJump)
            reachable += dp[i-minJump];

        if(i > maxJump)
            reachable -= dp[i-maxJump-1];

        dp[i] =
            reachable > 0 &&
            s[i] === '0';
    }

    return dp[n-1];
};

File: 2026/05-May/25-Monday/solution.py
class Solution:
    def canReach(self, s: str,
                 minJump: int,
                 maxJump: int) -> bool:

        n = len(s)

        dp = [False]*n
        dp[0] = True

        reachable = 0

        for i in range(1,n):

            if i >= minJump:
                reachable += dp[i-minJump]

            if i > maxJump:
                reachable -= dp[i-maxJump-1]

            dp[i] = reachable > 0 and s[i]=='0'

        return dp[-1]
